system_metrics_collector/src/system_metrics_collector/utilities.cpp

```diff
@@ -85,6 +87,28 @@ ProcCpuData ProcessStatCpuLine(const std::string & stat_cpu_line)
   return parsed_data;
 }
 
+ProcPidCpuData MeasurePidCpuTime()
+{
+  static const int kNumCpus = sysconf(_SC_NPROCESSORS_ONLN);
+  timespec process_time{};
+  timespec monotonic_time{};
+
+  if (clock_gettime(CLOCK_PROCESS_CPUTIME_ID, &process_time) != 0) {
+    RCUTILS_LOG_ERROR_NAMED("MeasurePidCpuTime", "unable to get process cpu time");
+    return ProcPidCpuData();
+  }
+  if (clock_gettime(CLOCK_MONOTONIC_RAW, &monotonic_time) != 0) {
+    RCUTILS_LOG_ERROR_NAMED("MeasurePidCpuTime", "unable to get monotonic cpu time");
+    return ProcPidCpuData();
+  }
+
+  ProcPidCpuData measured_data;
```

Comment:
nit: you could move this to line 95 since it is default initialized to empty data.

Reply:
I put this here to avoid that initialization until we know both measurement calls to `clock_gettime()` are successful and we've passed those early returns.

Reply:
Is there a different in initialization? For any of the failure cases you're going to return a default initialized object with "empty" data fields. If the two checks pass and we don't return the empty object, the fields are now set.

Reply:
Okay, I see what you mean.

Reply:
Not a blocker though. I think this is g2g.

Reply:
It's done now.